opentechinstitute/commotion-router-test-suite
# Python3

Summary: Update test scripts from python 2.7 to python 3. Minor code cleanup, but no real refactoring or change in function.

To test:
1. If necessary, install python 3. If you're already running 2.7 you may have to jump through some hoops to get both working on the same system.
2. From the repo root, run `py.test`. The tests should error out with the following exception: "UnboundLocalError: local variable 'commotion_client_ip' referenced before assignment".
3. Connect to a commotion AP.
4. Run `py.test` again. The test scripts should run with the following results: 1 failed, 4 passed, 1 xfailed.
5. Run `py.test tests/test_commotion_router_user.py` and `py.test tests/commotion_router_admin.py`. You should get successes and failures on the same tests as in step 4, but only the specified tests will run.
